### PR TITLE
Update buildcff2vf

### PR DESCRIPTION
**Pull request by readroberts**
Address issue 816: wrong set of masters referenced when fixing compatibility issues, if default master is not the first font the master font list.
Address issue 817: fix unintentional filtering of features when using the subsetting option.

### CODE REVIEW

File: python/afdko/buildcff2vf.py [hunk 1]
```diff
@@ -156,14 +152,12 @@ def make_curve_coords(self, coords, is_default):
         if is_default:
             new_coords = []
             for i, cur_coords in enumerate(coords):
```
Comment (cjchapman) [inline python/afdko/buildcff2vf.py]: You're not using the `i` anymore, so please change this to:
```python
for cur_coords in coords:
```

File: python/afdko/buildcff2vf.py [hunk 2]
```diff
@@ -156,14 +152,12 @@ def make_curve_coords(self, coords, is_default):
         if is_default:
```
Comment (cjchapman) [inline python/afdko/buildcff2vf.py]: You're not using `prev_coords` below, so you don't need `prev_cmd` in line 151 above.

File: python/afdko/buildcff2vf.py [hunk 3]
```diff
@@ -156,14 +152,12 @@ def make_curve_coords(self, coords, is_default):
...
             new_coords = []
             for i, cur_coords in enumerate(coords):
-                prev_coords = prev_cmd[1][i]
-                master_coords = self.make_flat_curve(prev_coords[:2],
-                                                     cur_coords)
+                master_coords = self.make_flat_curve(cur_coords)
                 new_coords.append(master_coords)
         else:
             cur_coords = coords
             prev_coords = prev_cmd[1][-1]
```
Comment (cjchapman) [inline python/afdko/buildcff2vf.py]: You're not using `prev_coords`, so this line can be removed (as can line 151 above).

File: python/afdko/buildcff2vf.py [hunk 4]
```diff
@@ -297,6 +291,10 @@ def getCharStrings(self, num_masters, private=None, globalSubrs=None):
             charString = T2CharString(
                 program=program, private=private, globalSubrs=globalSubrs)
             t2List.append(charString)
+        # if default_idx is not 0, we need to move it to the right index.
+        if default_idx:
+            default_font_cs = t2List.pop(0)
+            t2List.insert(default_idx, default_font_cs)
```
Comment (cjchapman) [inline python/afdko/buildcff2vf.py]: Please remove the trailing whitespace on line 297.

File: python/afdko/buildcff2vf.py [hunk 5]
```diff
@@ -297,6 +291,10 @@ def getCharStrings(self, num_masters, private=None, globalSubrs=None):
             charString = T2CharString(
                 program=program, private=private, globalSubrs=globalSubrs)
             t2List.append(charString)
+        # if default_idx is not 0, we need to move it to the right index.
+        if default_idx:
+            default_font_cs = t2List.pop(0)
+            t2List.insert(default_idx, default_font_cs) 
         return t2List
```
Comment (cjchapman) [inline python/afdko/buildcff2vf.py]: One more thing: `glyphOrder` is not used in `_get_cs()` starting at line 301 below:

**Comment by cjchapman** [comment]
Also, the flake8 report has a few things from above and a few new things:
```
$ flake8 python/afdko/buildcff2vf.py 
python/afdko/buildcff2vf.py:146:52: E226 missing whitespace around arithmetic operator
python/afdko/buildcff2vf.py:146:54: E226 missing whitespace around arithmetic operator
python/afdko/buildcff2vf.py:146:72: E226 missing whitespace around arithmetic operator
python/afdko/buildcff2vf.py:146:74: E226 missing whitespace around arithmetic operator
python/afdko/buildcff2vf.py:159:13: F841 local variable 'prev_coords' is assigned to but never used
python/afdko/buildcff2vf.py:258:80: E501 line too long (89 > 79 characters)
python/afdko/buildcff2vf.py:297:56: W291 trailing whitespace
python/afdko/buildcff2vf.py:336:80: E501 line too long (84 > 79 characters)
```

**Comment by readroberts** [comment]
@cjchapman @miguelsousa I fixed the Codacy and flake8 issues. I see the first two builds failed, but it looks like it is an issue with the build environment rather than the code.

**Comment by cjchapman** [comment]
@readroberts (cc: @miguelsousa) this branch was trying to use the `miguelsousa` fork of `cibuildwheel` which no longer exists, so I've cherry-picked the change back to the `adobe-type-tools` fork from the `develop` branch to this branch.

**Comment by cjchapman** [comment]
@readroberts (cc: @miguelsousa) Correction: I've rebased this branch off the `develop` branch to pick up the change. (Sorry, I was spaced, I really shouldn't look at GitHub this late in the day.)

**Comment by miguelsousa** [comment]
I've reworked the commits. Should be ready for @josh-hadley to review.

**Pull request merged**